src/main/java/com/jcraft/jsch/OpenSshCertificateUtil.java

```diff
@@ -0,0 +1,216 @@
+      throws IllegalArgumentException {
+    return extractSpaceDelimitedString(certificateFileContent, 1);
+  }
+
+  /**
+   * A private utility method to safely extract a space-delimited string from a certificate content
+   * string at a given index. This method is null-safe and handles out-of-bounds indices gracefully
+   * by returning {@code null}.
+   *
+   * @param certificate The string to be parsed, typically representing certificate content.
+   * @param index The zero-based index of the field to extract.
+   * @return The string field at the specified index, or {@code null} if the input is invalid or the
+   *         index is out of bounds.
+   */
+  public static String extractSpaceDelimitedString(String certificate, int index) {
+    if (certificate == null || certificate.trim().isEmpty()) {
+      return null;
+    }
+    String[] fields = certificate.split("\\s+");
+
+    if (index >= 0 && index < fields.length) {
+      return fields[index];
+    } else {
+      return null;
+    }
+  }
+
+
+  /**
+   * Extracts the key type from encoded key data provided as a byte array. Converts the byte array
+   * to a UTF-8 string and delegates to the string version of this method.
+   *
+   * @param s the encoded key data as a byte array, may be null
+   * @return the key type as a string, or an empty string if the input is null/empty
+   * @throws IllegalArgumentException if the data format is invalid (no space delimiter found)
+   * @see #extractKeyType(String)
+   */
+  public static String extractKeyType(byte[] s) throws IllegalArgumentException {
+    String str = tabToSpaceAndTrim(s);
+    return extractKeyType(str);
+  }
+
+
+  /**
+   * Determines whether the given {@link OpenSshCertificate} is valid at the current local system
+   * time.
+   *
+   * @param cert to check
+   * @return {@code true} if the certificate is valid according to its timestamps, {@code false}
+   *         otherwise
+   */
+  static boolean isValidNow(OpenSshCertificate cert) {
+    long now = MILLISECONDS.toSeconds(System.currentTimeMillis());
+    return Long.compareUnsigned(cert.getValidAfter(), now) <= 0
+        && Long.compareUnsigned(now, cert.getValidBefore()) < 0;
+  }
+
+  /**
+   * Converts a Unix timestamp to a {@link Date} string representation.
+   *
+   * If the timestamp is negative, it indicates an infinite expiration time, and the method returns
+   * the string "infinity". Otherwise, it converts the timestamp from seconds to milliseconds and
+   * returns the default string representation of the resulting {@link Date} object.
+   *
+   * @param timestamp The Unix timestamp in seconds.
+   * @return A string representing the date, or "infinity" if the timestamp is negative.
+   */
+  static String toDateString(long timestamp) {
+    if (timestamp < 0) {
+      return "infinity";
+    }
+    Date date = new Date(TimeUnit.SECONDS.toMillis(timestamp));
+    return date.toString();
```

Comment:
Can we just use `SftpATTRS.toDateString()`?
We can extract that method out of `SftpATTRS` and put it into a util class btw.